src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,190 @@
+        sortedSections.add(firstSection);
+        addNotFirstSections(sortedSections, firstSection);
+        return sortedSections;
+    }
+
+    private Section getFirstSection() {
+        return sections.stream()
+                .filter(this::isFirst)
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(CAN_NOT_FIND_FIRST_SECTION));
+    }
+
+    private boolean isFirst(Section section) {
+        return sections.stream()
+                .filter(it -> !section.equals(it))
+                .noneMatch(section::isNext);
+    }
+
+    private void addNotFirstSections(List<Section> sortedSections, Section before) {
+        Optional<Section> optionalNext = getNextSection(before);
+        if (!optionalNext.isPresent()) {
+            return;
+        }
+        Section next = optionalNext.get();
+        sortedSections.add(next);
+        addNotFirstSections(sortedSections, next);
+    }
+
+    private Optional<Section> getNextSection(Section section) {
+        return sections.stream()
+                .filter(it -> it.isNext(section))
+                .findFirst();
+    }
+
+    public void remove(Station station) {
+        checkRemovable(station);
+        Section firstSection = getFirstSection();
+        Section lastSection = getLastSection();
+        if (firstSection.isUpStation(station)) {
+            this.sections.remove(firstSection);
+            return;
+        }
+        if (lastSection.isDownStation(station)) {
+            this.sections.remove(lastSection);
+            return;
+        }
+        removeMiddleSection(station);
+    }
+
+    private void checkRemovable(Station station) {
+        if (isInvalidSectionsSize()) {
+            throw new IllegalArgumentException(SECTIONS_HAS_ONLY_ONE);
+        }
+        if (!getSortedStation().contains(station)) {
+            throw new IllegalArgumentException(NOT_CONTAINS_STATION);
+        }
+    }
+
+    private Section getLastSection() {
+        return sections.stream()
+                .filter(this::isDown)
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("can not find section."));
+    }
+
+    private boolean isDown(Section section) {
+        return sections.stream()
+                .filter(it -> !section.equals(it))
+                .noneMatch(section::isBefore);
+    }
+
+    private boolean isInvalidSectionsSize() {
+        return sections.size() < REMOVABLE_SIZE;
+    }
+
+    private void removeMiddleSection(Station station) {
+        Section beforeSection = getBeforeSection(station);
+        Section afterSection = getAfterSection(station);
+        addCombinationSection(beforeSection, afterSection);
+        removeBeforeAfterSection(beforeSection, afterSection);
+    }
+
+    private Section getBeforeSection(Station station) {
+        return sections.stream()
+                .filter(section -> section.getDownStation().equals(station))
```

Comment:
```suggestion
                .filter(section -> section.isSameDownStation(station))
```
객체를 넘겨서 비교하도록 시킨다면 `get`을 사용하지 않을수 있지 않을까요?!